src/neo/Network/P2P/Peer.cs

```diff
@@ -207,26 +210,27 @@ private void OnStart(ChannelsConfig config)
         /// <param name="local">The local endpoint of TCP connection.</param>
         private void OnTcpConnected(IPEndPoint remote, IPEndPoint local)
         {
-            ImmutableInterlocked.Update(ref ConnectingPeers, p => p.Remove(remote));
```

Comment:
The `remote` should be removed from `ConnectingPeers` even if it is disconnected.

Reply:
Done